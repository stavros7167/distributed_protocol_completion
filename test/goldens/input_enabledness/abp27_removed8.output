Parsing file test/goldens/input_enabledness/abp27_removed8.txt
# automata read: 11
SOLUTION #1 FOUND:
** Solution 1 **
('receiver', 'r0', "p1'?", 'r5')
('receiver', 'r3', "p0'?", 'r2')
('sender', 's0', "a1'?", 's0')
('sender', 's0', 'timeout?', 's0')
('sender', 's2', "a1'?", 's1')
('sender', 's2', 'timeout?', 's1')
('sender', 's4', "a0'?", 's4')
('sender', 's4', 'timeout?', 's4')
('sender', 's6', "a0'?", 's5')
('sender', 's6', 'timeout?', 's5')
{'candidate transitions': 76, 'iterations': 78}
