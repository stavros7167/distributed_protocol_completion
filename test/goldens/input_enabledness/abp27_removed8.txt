// re-running the synthesis experiments from scratch
// we now include input-enabledness to the protocol processes
// we also include output-fairness for the protocol processes


// ENVIRONMENT PROCESSES

include "channels.txt"

forward_channel = channel(p0, p1, p0', p1')

backward_channel = channel(a0, a1, a0', a1')

// include "client_sending.txt"
include "client_sending_input_enabled.txt"

include "client_receiving_without_fairness.txt"

include "timer.txt"


// INCOMPLETE PROCESSES FOR SYNTHESIS

include "incomplete_sender_with_fairness_and_inputenabledness_removed8.txt"

include "incomplete_receiver_with_fairness_and_inputenabledness.txt"

// PROPERTIES

include "safety_monitor.txt"

include "safety_monitor2.txt"

include "liveness1.txt"  // [](send -> <> deliver)

include "liveness3.txt"  // [](send -> <> done)

// include "infinitely_often_send_input_complete.txt"

// strong_non_blocking [send, deliver, done, timeout, p0, p1, a0, a1, p0', p1', a0', a1']

